pkg/git/git.go

```diff
@@ -0,0 +1,80 @@
+package git
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	giturls "github.com/whilp/git-urls"
+)
+
+func run(args ...string) error {
+	cmd := exec.Command("git", args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func HasStagedChange() error {
+	err := run("diff", "--staged", "--exit-code")
+	if err != nil {
+		return errors.Errorf("repository contains staged changes: %q", err)
+	}
+	return nil
+}
+
+func RmRecursive(path string) error {
+	return run("rm", "-r", path)
+}
+
+func AddAll(path string) error {
+	return run("add", "-A", path)
+}
+
+func SubtreeAdd(path, repository, revision string) error {
+	return run("subtree", "add", "--squash", "--prefix", path, repository, revision)
+}
+
+func Commit(message string) error {
+	log.Info("Committing the changes...")
+	err := run("commit", "-m", message)
+	if err == nil {
+		log.Info("Committed the changes.")
+	}
+	return err
+}
+
+func Push() error {
+	log.Info("Pushing to the remote...")
+	err := run("push")
```

Comment:
behavior of `git push` varies depending on the value of `push.default` (see `man git-config`). In particular, doing simply `git push` may either fail (if there's no tracking information in your current branch) or push unwanted branches.

Do this to ensure that the current branch is pushed to the `origin` remote branch with the same name (see `man git-push`):

```suggestion
	err := run("push", "origin", "HEAD")
```

Reply:
fixed